process/block/shardblock_test.go

```diff
@@ -4558,3 +4558,64 @@ func TestShardProcessor_RequestMetaHeadersIfNeededShouldAddHeaderIntoTrackerPool
 	expectedAddedNonces := []uint64{6, 7}
 	assert.Equal(t, expectedAddedNonces, addedNonces)
 }
+
+func TestShardProcessor_CheckEpochCorrectnessShouldRemoveAndRequestStartOfEpochMetaBlockWhenEpochDoesNotMatch(t *testing.T) {
+	t.Parallel()
+
+	removeHeaderByHashWasCalled := false
+	requestMetaHeaderWasCalled := false
+	epochStartMetaHash := []byte("epoch start meta hash")
+
+	currentHeader := &block.Header{
+		Epoch: uint32(1),
+	}
+	nextHeader := &block.Header{
+		Epoch:              uint32(2),
+		EpochStartMetaHash: epochStartMetaHash,
+	}
+
+	blockChainMock := &mock.BlockChainMock{
+		GetCurrentBlockHeaderCalled: func() data.HeaderHandler {
+			return currentHeader
+		},
+	}
+	epochStartTriggerStub := &mock.EpochStartTriggerStub{
+		MetaEpochCalled: func() uint32 {
+			return currentHeader.Epoch
+		},
+	}
+	poolsHolderStub := &dataRetrieverMock.PoolsHolderStub{
+		HeadersCalled: func() dataRetriever.HeadersPool {
+			return &mock.HeadersCacherStub{
+				RemoveHeaderByHashCalled: func(headerHash []byte) {
+					if bytes.Equal(headerHash, epochStartMetaHash) {
+						removeHeaderByHashWasCalled = true
+					}
+				},
+			}
+		},
+	}
+	requestHandlerStub := &testscommon.RequestHandlerStub{
+		RequestMetaHeaderCalled: func(headerHash []byte) {
+			if bytes.Equal(headerHash, epochStartMetaHash) {
+				requestMetaHeaderWasCalled = true
+			}
+		},
+	}
+
+	coreComponents, dataComponents, bootstrapComponents, statusComponents := createComponentHolderMocks()
+	dataComponents.BlockChain = blockChainMock
+	dataComponents.DataPool = poolsHolderStub
+	arguments := CreateMockArguments(coreComponents, dataComponents, bootstrapComponents, statusComponents)
+	arguments.EpochStartTrigger = epochStartTriggerStub
+	arguments.RequestHandler = requestHandlerStub
+	sp, _ := blproc.NewShardProcessor(arguments)
+
+	err := sp.CheckEpochCorrectness(nextHeader)
+
+	time.Sleep(100 * time.Millisecond)
```

Comment:
can use a channel here:
On L4597 you declare a channel:
```GO
ch := make(chan struct{})
```

On L4601 you close the dummy channel:
```GO
close(ch)
```

Here, on L4616 you can have:
```GO
select{
case <- ch:
case <- time.After(time.Minute):
    assert.Fail(t, "timeout while waiting the sending of the request for the meta header")
}
```

Reply:
Ok, done